File: cpp/section05/exercises/two_sum.cpp
#include <iostream>
#include <vector>
#include <sstream>

std::string printVector (const std::vector<int> & vector) {    

    int i;
    int j;

    if (vector[0] && vector[1]) {
       std::ostringstream oss;
       oss << "{" << vector[0] << " , " << vector[1] << "}";
        return oss.str();
    }

    return "{}";
}

std::vector<int> hasPairWithSumBruteForce (const std::vector<int> & vector, int sum) {
    for (int i = 0; i < vector.size(); i++) {
        for (int j = i + 1; j < vector.size(); j++) {
            if (vector[i] + vector[j] == sum) {
                return {i,j};
            }
        }
    }
    return {};
}

int main () {
    std::vector<int> vector1 = {1,2,4,4};

    std::string result = printVector(hasPairWithSumBruteForce(vector1,8));
    
    std::cout << "Result: "  << result << std::endl;


    return 1;
}

File: java/DSA/src/main/java/section05/exercises/TwoSum.java
package section05.exercises;

public class TwoSum {
    public static void main (String [] args) {
        int [] sortedArray = {1,2,3,9};
        int [] result = hasPairWithSumBruteForce(sortedArray,5);
        System.out.println("{" + result[0] + "," + result[1]+"}");
    }


    public static int [] hasPairWithSumBruteForce (int [] array,int sum) {
        for (int i = 0 ; i < array.length; i++) {
            for (int j = 0; j < array.length; j++) {
                if (array[i] + array[j] == sum) {
                    return  new int [] {i,j};
                }
            }
        }
        return new int[] {};
    }
}
